survey-runner-application/cloudwatch.tf

```diff
@@ -0,0 +1,103 @@
+  dimensions {
+    AutoScalingGroupName = "${aws_elastic_beanstalk_environment.survey_runner_prime.autoscaling_groups[0]}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "5xx_errors" {
+  alarm_name = "${var.env}-survey-runner-5xx-errors"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods = "1"
+  metric_name = "HTTPCode_Backend_5XX"
+  namespace = "AWS/ELB"
+  period = "60"
+  statistic = "Sum"
+  threshold = "1"
+  alarm_description = "There have been at least 1 5xx errors in the past 60 seconds"
+  alarm_actions       = ["arn:aws:sns:eu-west-1:${data.aws_caller_identity.current.account_id}:${var.env}-slack-alarm"]
+  dimensions {
+    LoadBalancerName = "${aws_elastic_beanstalk_environment.survey_runner_prime.load_balancers[0]}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "4xx_errors" {
+  alarm_name = "${var.env}-survey-runner-4xx-errors"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods = "2"
+  metric_name = "HTTPCode_Backend_4XX"
+  namespace = "AWS/ELB"
+  period = "60"
+  statistic = "Sum"
+  threshold = "100"
+  alarm_description = "There have been at least 100 4xx errors in the past 120 seconds"
+  dimensions {
+    LoadBalancerName = "${aws_elastic_beanstalk_environment.survey_runner_prime.load_balancers[0]}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "max_host_count" {
+  alarm_name = "${var.env}-survey-runner-max_host_count"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods = "1"
+  metric_name = "HealthyHostCount"
+  namespace = "AWS/ELB"
+  period = "60"
+  statistic = "Maximum"
+  threshold = "${var.eb_max_size}"
+  alarm_description = "The number of ElasticBeanstalk instances is at its Maximum of ${var.eb_max_size}"
+  alarm_actions       = ["arn:aws:sns:eu-west-1:${data.aws_caller_identity.current.account_id}:${var.env}-slack-alarm"]
+  dimensions {
+    LoadBalancerName = "${aws_elastic_beanstalk_environment.survey_runner_prime.load_balancers[0]}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "min_host_count" {
+  alarm_name = "${var.env}-survey-runner-min_host_count"
+  comparison_operator = "LessThanThreshold"
+  evaluation_periods = "1"
+  metric_name = "HealthyHostCount"
+  namespace = "AWS/ELB"
+  period = "60"
+  statistic = "Minimum"
+  threshold = "${var.eb_min_size}"
+  alarm_description = "The number of ElasticBeanstalk instances is below its Minimum of ${var.eb_min_size}"
+  alarm_actions       = ["arn:aws:sns:eu-west-1:${data.aws_caller_identity.current.account_id}:${var.env}-slack-alarm"]
+  dimensions {
+    LoadBalancerName = "${aws_elastic_beanstalk_environment.survey_runner_prime.load_balancers[0]}"
+  }
+}
+
+resource "aws_cloudwatch_log_metric_filter" "no_database_connections_remaining" {
+  name           = "no_remaining_database_connections"
+  pattern        = "remaining connection slots are reserved for non-replication superuser connections"
```

Comment:
This never matches because of the `-` in non-replication:
> Metric filter terms that include characters other than alphanumeric or underscore must be placed inside double quotes ("").

from https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html

Using `"\"remaining connection slots are reserved for non-replication superuser connections\""` fixes it.